File: assignment1-3/src/main/test_wc.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"regexp"
	"strings"
)

func main() {
	fmt.Printf("Searching for word: \"%v\" in %v\n", os.Args[1], os.Args[2:])
	var count int
	for _, file := range os.Args[2:] {
		text, _ := ioutil.ReadFile(file)
		words := strings.Fields(string(text))
		reg, _ := regexp.Compile("[^0-9a-zA-Z]+")
		for _, word := range words {
			validWord := reg.ReplaceAllString(word, "")
			if validWord == os.Args[1] {
				// fmt.Printf("%v\n", word)
				count++
			}
		}
	}
	fmt.Printf("%v: %v\n", os.Args[1], count)
}

File: assignment1-3/src/main/test_wc.py
import re
import sys

print("Searching for word: \"{}\" in {}".format(sys.argv[1], sys.argv[2:]))
count = 0
for f in sys.argv[2:]:
    with open(f, 'r') as fd:
        text = fd.read()
        for word in text.split():
            validWord = re.sub(r'\W+', '', word)
            validWord = validWord.replace("_", "")
            if validWord == sys.argv[1]:
                # print(word)
                count += 1
print("{}: {}".format(sys.argv[1], count))
